compiler/passes/convert-uast.cpp

```diff
@@ -4366,12 +4367,16 @@ struct Converter {
 };
 
 bool LoopAttributeInfo::insertGpuEligibilityAssertion(BlockStmt* body) {
+  bool inserted = false;
   if (assertOnGpuAttr) {
-    body->insertAtTail(new CallExpr(PRIM_ASSERT_ON_GPU,
-                                    new SymExpr(gTrue)));
-    return true;
+    body->insertAtTail(new CallExpr("chpl__assertOnGpuAttr"));
```

Comment:
I know I said this wouldn't bother me, but could you explain to me (or point me to an issue) why this is better?

Reply:
Some of my reasons:

1. I can use the overload resolution process to select candidate functions. For instance, I may use a where clause to avoid inserting GPU primitives when CHPL_LOCALE_MODEL=flat
2. I can ensure that the arguments to the primitives are correctly typed (since calls are just “regular functions”).
3. I don’t have to change compiler code to adjust the behavior of attributes, to some extent
4. Using an experimental annotation (which I plan to work on next after this PR) I can get `chpldoc` to use these function definitions (like `chpl__assertOnGpuAttr`) to generate documentation for attributes, which addresses https://github.com/chapel-lang/chapel/issues/23149.

Reply:
> Using an experimental annotation (which I plan to work on next after this PR) I can get chpldoc to use these function definitions (like chpl__assertOnGpuAttr) to generate documentation for attributes, which addresses

Nice.

Thanks for explanation. You are right on all counts.